### ISSUE TITLE
Help Required to Understand Model Inputs

### ISSUE THREAD

**Issue by anujdutt9** [openvinotoolkit/training_extensions#532]
Hi. Thanks for a great framework. 
I have one thing that I don't understand though. If I take a model from here or from `pytorchcv`, define that in a config file, and define any input size, how is the model able to take in any input size for training/fine-tuning without any resize layer at input?
For example, if I load a pre-trained ShuffleNet model defined in `pytorchcv`, usually models are trained on 224x224 or 256x256 input images. But now if I want to fine-tune this pretrained model on my custom dataset, say of input size 1088x320, then how is this model able to fine-tune on these images without a resize layer at the input. I tried looking in the code but fail to understand that how a pretrained model can be fine tuned on any arbitrary sized images without a resize later at input.
Please help me understand this.
Thanks

**Comment by morkovka1337** [comment]
Hi.

Actually there is no any trick here. The thing is, most of models use convolutional layers and pooling layers, which are **insensible** to the input size. For example, when the convolutional kernel is applied to the image 5x5 with filter size 3, stride 1 and without padding (the simplest case) you will result in the feature map of size 3x3:
```
In [1]: import torch
In [2]: conv = torch.nn.Conv2d(1, 3, 3)

In [3]: conv
Out[3]: Conv2d(1, 3, kernel_size=(3, 3), stride=(1, 1))


In [3]: sample_input = torch.randn(1, 1, 5, 5)


In [4]: conv(sample_input).shape
Out[4]: torch.Size([1, 3, 3, 3])

```

If you, instead, feed input image in shape of 10x10, you will result in feature map of size 8x8. Look in the documentation, section [shape](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html) for details. The same thing is with Pooling layers (max pooling, average pooling).

Let's look at the official pytorch [implementation of Resnet50](https://pytorch.org/vision/stable/_modules/torchvision/models/resnet.html#resnet50)
It consists of (along the convolutional and pooling layers) AdaptiveAvgPool2d and FC layers. Fc layer takes as an input tensor of an exact shape, but how it is achieved? The reason is adaptive AdaptiveAvgPool2d layer. If you look at the [documentation](https://pytorch.org/docs/stable/generated/torch.nn.AdaptiveAvgPool2d.html), you will see that it returns Tensor of specified shape **for any input**, [automatically adjusting kernel](https://stackoverflow.com/questions/53841509/how-does-adaptive-pooling-in-pytorch-work).

Again, there is no implicit reshape, some layers are insensible to the input size (like `conv` and `pool`), and some layers produce feature map of fixed shape for any input. If the network is designed such way that it tensor of specific shape, and got another shape, it will fail until you make extra tensor transformations.

Good luck!

**Comment by anujdutt9** [comment]
Hi, @morkovka1337 Thanks a lot for the great explanation and the code examples. This clears my understanding. 💯
